pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -388,17 +394,37 @@ protected BrokerResponse doHandleRequest(long requestId, String query, SqlNodeAn
     BrokerRequest brokerRequest = CalciteSqlCompiler.convertToBrokerRequest(pinotQuery);
     BrokerRequest serverBrokerRequest =
         serverPinotQuery == pinotQuery ? brokerRequest : CalciteSqlCompiler.convertToBrokerRequest(serverPinotQuery);
-    AuthorizationResult authorizationResult = accessControl.authorize(requesterIdentity, serverBrokerRequest);
 
-    _brokerMetrics.addPhaseTiming(rawTableName, BrokerQueryPhase.AUTHORIZATION,
-        System.nanoTime() - compilationEndTimeNs);
+    TableRouteProvider routeProvider;
 
-    if (!authorizationResult.hasAccess()) {
-      throwAccessDeniedError(requestId, query, requestContext, tableName, authorizationResult);
+    if (logicalTable != null) {
+      AuthorizationResult authorizationResult =
+          hasTableAccess(requesterIdentity, logicalTable.getPhysicalTableConfigMap().keySet(), requestContext,
+              httpHeaders);
+      if (!authorizationResult.hasAccess()) {
+        throwAccessDeniedError(requestId, query, requestContext, tableName, authorizationResult);
```

Comment:
Is it possible to know for which table access failed and send that in the error message to user?

Reply:
Is it actionable ? An analyst may not know that a logical table is being queried. Maybe the query is coming from a dashboard or a UI client. Also they may not have the permissions to change it.

Reply:
Having an error with logical table would users to seek permission for the logical table. Do we want that?

For the logical table, user should have access to the underline physical tables and that would would not appear directly to the user unless documentation/expert opinion is consulted. How can we reduce that friction?

Reply:
I've filed #15784 to debate the open questions more thoroughly.